apps/studio.giselles.ai/app/(main)/apps/components/searchable-agent-list.tsx

```diff
@@ -0,0 +1,290 @@
+"use client";
+
+import { formatTimestamp } from "@giselles-ai/lib/utils";
+import {
+	ArrowDownAZ,
+	ArrowUpAZ,
+	Clock,
+	LayoutGrid,
+	LayoutList,
+	Search,
+} from "lucide-react";
+import Link from "next/link";
+import { useEffect, useState } from "react";
+import { Button } from "@/components/ui/button";
+import {
+	DropdownMenu,
+	DropdownMenuContent,
+	DropdownMenuItem,
+	DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { Input } from "@/components/ui/input";
+import { AgentGrid } from "./agent-grid";
+import { DeleteAgentButton } from "./delete-agent-button";
+import { DuplicateAgentButton } from "./duplicate-agent-button";
+import type { AgentGridProps } from "./types";
+
+type SortOption = "name-asc" | "name-desc" | "date-desc" | "date-asc";
+type ViewMode = "grid" | "list";
+
+function ListItem({
+	agent,
+	isFirst,
+}: {
+	agent: AgentGridProps["agents"][0];
+	isFirst: boolean;
+}) {
+	const [relativeTime, setRelativeTime] = useState("");
+
+	useEffect(() => {
+		const update = () => {
+			setRelativeTime(
+				formatTimestamp.toRelativeTime(new Date(agent.updatedAt).getTime()),
+			);
+		};
+		update();
+		const id = setInterval(update, 60_000);
+		return () => clearInterval(id);
+	}, [agent.updatedAt]);
+
+	return (
+		<Link
+			href={`/workspaces/${agent.workspaceId}`}
+			className={`group flex items-center justify-between px-2 py-3 ${
+				!isFirst ? "border-t-[0.5px] border-white/10" : ""
+			}`}
```

Comment:
I want to implement isFirst using tailwindcss's https://tailwindcss.com/docs/hover-focus-and-other-states#first-last-odd-and-even without passing it as props.